client/v3/snapshot/v3_snapshot.go

```diff
@@ -46,7 +46,7 @@ func hasChecksum(n int64) bool {
 // Etcd <v3.6 will return "" as version.
 func SaveWithVersion(ctx context.Context, lg *zap.Logger, cfg clientv3.Config, dbPath string) (version string, err error) {
 	if lg == nil {
-		lg = zap.NewExample()
+		lg = zap.NewNop()
```

Comment:
Why do you prefer Nop ?

Please add a comment, otherwise it's tempting during debugging to add back 'any logging'.

Reply:
This is based on "no logging is better than inconsistent one. Alternatively I can remove the `nil` check.